main.cpp

```diff
@@ -8,9 +8,10 @@ int main()
     try
     {
         auto textProcessor = RabinKarpSearch<StandardHash>("../test.txt", "ipsum");
-        std::vector<int> matches = textProcessor.search();
+        std::vector<size_t> matches = textProcessor.search();
 
-        for (const int &i : matches) {
+
+        for (auto i : matches) {
```

Comment:
`auto` looks much cleaner, but I like to stick with static typing, isn't `int` a good call too? It's up to you, but I'm curious why it was chosen :smile:

Reply:
Auto is almost always used with for each loops in cpp. Additionally using reference for simple datatypes as int is not a good practice. It's better to copy them.